pkg/asset/manifests/gcp/cluster.go

```diff
@@ -0,0 +1,150 @@
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(enableFlowLogs),
+		Purpose:        ptr.To(defaultSubnetCreationPurpose),
+	}
+
+	workerSubnet := installConfig.Config.GCP.ComputeSubnet
+	workerSubnetDescription := "Compute Subnet used by the installer"
+	if workerSubnet == "" {
+		workerSubnet = subnetName(clusterID.InfraID, "worker")
+		workerSubnetDescription = "Compute Subnet owned by the installer"
+	}
+	worker := capg.SubnetSpec{
+		Name:           workerSubnet,
+		CidrBlock:      "",
+		Description:    ptr.To(workerSubnetDescription),
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(enableFlowLogs),
+		Purpose:        ptr.To(defaultSubnetCreationPurpose),
+	}
+
+	// Add the CIDR information.
+	machineV4CIDRs := []string{}
+	for _, network := range installConfig.Config.Networking.MachineNetwork {
+		if network.CIDR.IPNet.IP.To4() != nil {
+			machineV4CIDRs = append(machineV4CIDRs, network.CIDR.IPNet.String())
+		}
+	}
+
+	if len(machineV4CIDRs) == 0 {
+		return nil, fmt.Errorf("failed to parse machine CIDRs")
+	}
+
+	_, ipv4Net, err := net.ParseCIDR(machineV4CIDRs[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine network CIDR: %w", err)
+	}
+
+	if installConfig.Config.GCP.ControlPlaneSubnet == "" {
+		masterCIDR, err := cidr.Subnet(ipv4Net, 1, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the master subnet %w", err)
+		}
+		// The subnet must be created, provide the CIDR.
+		master.CidrBlock = masterCIDR.String()
+	}
+
+	if installConfig.Config.GCP.ComputeSubnet == "" {
+		workerCIDR, err := cidr.Subnet(ipv4Net, 1, 1)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the worker subnet %w", err)
+		}
+		// The subnet must be created, provide the CIDR.
+		worker.CidrBlock = workerCIDR.String()
+	}
+
+	subnets := []capg.SubnetSpec{master, worker}
+
+	labels := make(map[string]string, len(installConfig.Config.GCP.UserLabels)+1)
+	// add OCP default label.
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID.InfraID)] = "owned"
+	for _, label := range installConfig.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	gcpCluster := &capg.GCPCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		},
+		Spec: capg.GCPClusterSpec{
+			Project: installConfig.Config.GCP.ProjectID,
+			Region:  installConfig.Config.GCP.Region,
+			Network: capg.NetworkSpec{
+				Name:                  ptr.To(networkName),
+				AutoCreateSubnetworks: ptr.To(autoCreateSubnets),
+				Subnets:               subnets,
+			},
+			AdditionalLabels: labels,
+		},
+	}
+
+	manifests = append(manifests, &asset.RuntimeFile{
+		Object: gcpCluster,
+		File:   asset.File{Filename: "02_gcp-cluster.yaml"},
+	})
+
+	return &capiutils.GenerateClusterAssetsOutput{
+		Manifests: manifests,
+		InfrastructureRef: &corev1.ObjectReference{
+			APIVersion: "infrastructure.cluster.x-k8s.io/v1beta2",
```

Comment:
I think this should be `v1beta1` based on https://github.com/kubernetes-sigs/cluster-api-provider-gcp/tree/main/api
```suggestion
			APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
```